crypto/crypto.go

```diff
@@ -34,11 +35,11 @@ func Sign(key SigningKey, message []byte) []byte {
 	return ed25519.Sign(ed25519.PrivateKey(key), message)
 }
 
-func Verify(key SigningKey, message, sig []byte) bool {
-	pk, ok := ed25519.PrivateKey(key).Public().(ed25519.PublicKey)
-	if !ok {
-		return false
-	}
+func SigningKeyToPublic(key SigningKey) []byte {
+	return ed25519.PrivateKey(key).Public().(ed25519.PublicKey)
+}
+
+func Verify(pk, message, sig []byte) bool {
 	return ed25519.Verify(pk, message, sig)
 }
```

Comment:
LGTM. I see your changes also suggest that we make ``Sign`` a method of ``SigningKey``. That's interesting. I'll go and make the suggested changes to ``Verify`` in the ``str-tb-verify-idea`` branch so they're in there when @c633 merges is that branch with ``str-tb``.